### DISCUSSION TITLE
Nvidia Support

### DISCUSSION THREAD

**Discussion by constantius24** [mocaccinoOS/mocaccino#65] [General] [answered]
HI
Could you provide instruction  how to install nvidia properiaty driver for nvidia legacy for example 430 cards
and nvidia stable driver ( for examplae 700 series ) cards. I can find nvidia LTS package in repo luet.... but i know nothing about is it for..
best regards

**Comment by joostruis** [comment]
I could try adding the nvidia-drivers-390.147 series. Would that support your card?

https://www.nvidia.com/en-us/drivers/unix/legacy-gpu/

**Comment by constantius24** [comment]
Yes please and thanks in advance

**Reply by joostruis** [reply]
https://github.com/mocaccinoOS/desktop/commit/f985a4a914cfc49122f89819319a17680e91cfc3

**Reply by joostruis** [reply]
So what you can try:
```
luet repo update
luet install kernel-modules/nvidia-drivers-legacy-lts
```

Not 100% sure if needed but I would regenerate the initramfs:
`mocaccino-dracut --rebuild-all`

Then reboot. If correct this will blacklist nouveau and you do not have to mess around with anything.

**Comment by joostruis** [comment]
Hey there. Did you try installing what I wrote above? Did it work for you?

**Reply by constantius24** [reply]
Hi. I have change my mind and installed radeon hd8490. But your work is needed. Best regardd

**Reply by joostruis** [reply]
Oh that is too bad you didn't test it.

**Reply by constantius24** [reply]
Ok i will test it. Today evening. I will report it to you

**Comment by constantius24** [comment]
luet repo update
luet install kernel-modules/nvidia-drivers-legacy-lts
mocaccino-dracut --rebuild-all

1. Tested works perfectly on Nvidia GeForce GT 710 and GeForce GT 430.
2. mocaccino-dracut --rebuild-all - is needed. Please enclusure it luet upgrade tool
3. Next step instruction From Properiaty Driver to Neauveaux Driver.
4. Thanks

**Reply by joostruis** [reply]
There is no need for nouveau instructions. Just:
`luet uninstall kernel-modules/nvidia-drivers-legacy-lts`

This also removes the module blacklist file and reboot.
On next boot you are back on nouveau.

**Comment by joostruis** [comment]
https://github.com/mocaccinoOS/desktop/commit/fbaaadb2037f17c03675fe7e43b92f89da15eeaf

**Reply by joostruis** [reply]
INFO   🐚  Executing finalizer on  / sh [-c mocaccino-dracut --rebuild-all]
Yep works. Pushing to stable repo this fix.

**Comment by joostruis** [comment, marked as answer]
So to answer this question. Yes we now have support for legacy nvidia cards.
Check this page for supported cards: https://www.nvidia.com/en-us/drivers/unix/legacy-gpu/

To install do this:
```
luet repo update
luet install kernel-modules/nvidia-drivers-legacy-lts
```
This will now also update dracut on install. No need to mess around with a command anymore.

If you want to revert to nouveau:

`luet uninstall kernel-modules/nvidia-drivers-legacy-lts`

And reboot.

**Reply by 01fiore** [reply]
Hi, I have successfully implemented the Nvidia 470 drivers with a GTX 1050 Ti card, the 390 are also functional.

Great work, if I may, you should add Clementine in your repositories, thank you.

**Reply by joostruis** [reply]
Great that you got it working! 
Your other question is a bit offtopic but apps/clementine is available in the community repository.

**Comment by constantius24** [comment]
PS.: Installing Nvidia Legacy 390xx driver disable in GDM Wayland. 390 does not support it.

**Reply by constantius24** [reply]
after the last kernel upgrade to the 5/15/50 nvidia 470 driver crashed. Output no support for Nvidia 710 card. I have to uninstall it and install 390-legacy. But video and system run very slowly. ( Gnome ). It is strange because 470 support nvidia 710...

**Comment by constantius24** [comment]
after the last kernel upgrade to the 5/15/50 nvidia 470 driver crashed. Output no support for Nvidia 710 card. I have to uninstall it and install 390-legacy. But video and system run very slowly. ( Gnome ). It is strange because 470 support nvidia 710...
